---
title: "03 - Co-expression of astrocyte and interneuron programs in normal brain and tumor cells"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  html_document:
    keep_md: true
    theme: flatly
    css: ../../include/style.css
    toc: yes
    toc_depth: 4
    number_sections: true
    df_print: paged
    includes:
      before_body: ../../include/header.html
      after_body:  ../../include/footer.html
---

<!-- FRONT MATTER, insert configuration info -->

```{r header, echo = FALSE, results = 'asis', warning = FALSE}

# Index of the document
# ...determines name of the subfolder of `outputs` and `figures`
doc_id <- "03"
subdir <- "singlecell_normal"

suppressMessages(library(here))

# Knit child document with header
res <- knitr::knit_child(here("include", "header.Rmd"), envir = environment(), quiet = TRUE)
cat(res, sep = '\n')

```

<!-- END OF FRONT MATTER -->


# Overview

Given that we a dual neuronal-astrocytic gene expression program in the tumors,
here we interrogate the expression of interneuron and astrocytic
gene signatures in the normal brain, and then evaluate the same signatures
in the tumors, to understand if this is an
aberrant state, or also observed under normal conditions.

For each dataset:

1. Take the mean expression each of an interneuron gene signature and an astrocyte signature
2. Subset to cells which are astrocytes, interneurons, or radial glia
3. Generate a scatterplot of cells based on their expression of the two signatures

# Libraries

```{r libs, warning = FALSE, message = FALSE, echo = TRUE, cache = FALSE}

# Load liraries here
library(here)
library(tidyr)
library(dplyr)
library(glue)
library(ggplot2)
library(purrr)
library(readr)
library(Seurat)

```


# Load data

Human fetal brain (5-37 PCW, Nowakowski et al, Science, 2017):

```{r load_nowakowski, cache = FALSE}

load(here("reference_datasets/2017_Nowakowski/processed_data/nowakowski_seurat.Rda"))

```

Prepare the palette for this dataset, so that astrocytes, internurons, and
radial glial cells are distinct:

```{r pal_nowakowski}

palette_nowakowski <- read_csv(here("reference_datasets/2017_Nowakowski/processed_data/nowakowski2017_tableS4_cluster_labels_with_colours.csv")) %>%
  select(Cluster = `Cluster Name`, colour = Colour) %>%
  tibble::deframe()

palette_nowakowski["Astrocyte"] <- "#FF3F38"
palette_nowakowski[grepl("RG", names(palette_nowakowski))] <- "#ffcc00"

```

Mouse developing forebrain (E12-P6)

```{r load_jc, cache = FALSE}

load(here("reference_datasets/2019_Jessa/Jessa2019_forebrain_seurat.Rda"))

```

Prepare palette:

```{r pal_atlas}

palette_atlas <- read_tsv(here("reference_datasets/2019_Jessa/Jessa2019_Table_2a.tsv")) %>%
  select(Cluster, Colour) %>%
  mutate(Cluster = gsub("_", " ", Cluster)) %>%
  tibble::deframe()

palette_atlas[grepl("ASTR",      names(palette_atlas))] <- "#FF3F38"
palette_atlas[grepl("VRGC",      names(palette_atlas))] <- "#ffcc00"
palette_atlas[grepl("MGIN|CINH", names(palette_atlas))] <- "#084081"

```


# Analysis

## Get gene signatures

Here, we'll get the leading edge genes from the GSEA analysis, for one human
interneuron signature and one astrocyte signature, both enriched in G34R/V tumors.

```{r get_le}

dge_le <- read_tsv(here("bulk_transcriptome_epigenome/output/02/fgsea_results_padj<0.01_IDH_top_le.tsv"))

```


```{r signatures}

load(here("reference_datasets/2017_Nowakowski/processed_data/nowakowski_signatures.Rda"))

# Newborn interneuron signature
interneuron_genes    <- nowakowski_signatures$hg_sym$nIN1 
interneuron_genes_mm <- nowakowski_signatures$mm_sym$nIN1

# Astrocyte signature
astro_genes          <- nowakowski_signatures$hg_sym$Astrocyte
astro_genes_mm       <- nowakowski_signatures$mm_sym$Astrocyte

```


## Human fetal telencephalon (Nowakowski et al, 2017)

```{r nowakowski_prep}

data_nowakowski <- data.frame(
  interneuron_signature = cytobox::meanGeneExpression(nowa, interneuron_genes)$Mean_marker_expression,
  astrocyte_signature   = cytobox::meanGeneExpression(nowa, astro_genes)$Mean_marker_expression,
  cluster               = nowa@meta.data$WGCNAcluster,
  celltype              = nowa@meta.data$Cell_type
)

# Filter to cell types of interest for this analysis: astrocytes, inhibitory neurons
# and ganglionic eminence radial glia
data_nowakowski <- data_nowakowski %>% 
  filter(cluster %in% c("Astrocyte",
                        "nIN1", "nIN2", "nIN3", "nIN4",
                        "IN-CTX-CGE1", "IN-CTX-MGE1",
                        "MGE-IPC1", "MGE-IPC2", "MGE-IPC3",
                        "MGE-RG1", "MGE-RG2", "MGE-div"))

# Clean up
rm(nowa)

```

Generate the scatterplot:

```{r nowakowski_scatterplot, fig.width = 6, fig.height = 5, echo_fig = TRUE}

data_nowakowski %>% 
  rr_ggplot(aes(x = interneuron_signature, y = astrocyte_signature, colour = cluster), plot_num = 1) +
  geom_point(shape = 16, size = 1, alpha = 0.2) +
  scale_color_manual(values = palette_nowakowski, name = "Cell type") +
  geom_density_2d() +
  ggtitle("Human fetal telencephalon")

```



## Developing mouse forebrain (Jessa et al, 2019)

In the mouse atlas:

```{r mouse_forebrain_prep}

data_atlas <- data.frame(
  interneuron_signature = cytobox::meanGeneExpression(joint_cortex_small, interneuron_genes_mm)$Mean_marker_expression,
  astrocyte_signature   = cytobox::meanGeneExpression(joint_cortex_small, astro_genes_mm)$Mean_marker_expression,
  cluster               = gsub("_", " ", joint_cortex_small@meta.data$ID_20190730_with_blacklist_and_refined)
)

# Filter to relevant cell tpyes
data_atlas <- data_atlas %>% 
  filter(grepl("ASTR|VRGC|MGIN|CINH|INIP", cluster) & !grepl("BLACKLIST", cluster))

# Clean up
rm(joint_cortex_small)

```

Plot:

```{r mouse_forebrain_scatter, fig.width = 6, fig.height = 5, echo_fig = TRUE}

data_atlas %>% 
  rr_ggplot(aes(x = interneuron_signature, y = astrocyte_signature, colour = cluster), plot_num = 1) +
  geom_point(shape = 16, size = 1, alpha = 0.2) +
  scale_color_manual(values = palette_atlas, name = "Cell type") +
  geom_density_2d() +
  ggtitle("Normal mouse brain")

```

## H3G34R/V tumors

Load Seurat object with all samples combined, and re-constitute a V2 object from the V3 object:

```{r load_g34, cache = FALSE, eval = FALSE}

load("/lustre03/project/6004736/vlisi/fromHydra/SCRATCH/vlisi/LEGACY/G34/inHouseDataCombined/data/integrated_g34.seurat_v3.save_v2.Rda")

# get cell type projections used in Figure 5
load("/lustre03/project/6004736/vlisi/fromHydra/SCRATCH/vlisi/LEGACY/G34/allSamplesCombined/data/ScSnMetaDataWithNormals.RDat")
cell_meta <- ScSnDataWithNormals

# convert the object to V3
seurat_joint <- CreateSeuratObject(raw.data = integrated_g34@assays$RNA@counts,
                                   meta.data = integrated_g34@meta.data)
seurat_joint <- NormalizeData(seurat_joint)
seurat_joint <- SetDimReduction(seurat_joint, reduction.type = "umap", slot = "cell.embeddings",
                                integrated_g34@reductions[["umap"]]@cell.embeddings)
seurat_joint <- SetDimReduction(seurat_joint, reduction.type = "umap", slot = "key", "UMAP_")

# add in the cell type projections used in Figure 5
seurat_joint <- AddMetaData(seurat_joint, ScSnDataWithNormals[, c("ConsensusLineageRefinedConsensus"), drop = FALSE])
seurat_joint@meta.data$Cell_type_projection <- seurat_joint@meta.data$ConsensusHybrid2L3NP.smoothk10
seurat_joint@meta.data$Cell_type_broad_Fig5 <- seurat_joint@meta.data$ConsensusLineageRefinedConsensus
seurat_joint@meta.data$Status_normal_malignant <- seurat_joint@meta.data$CNVnormals

seurat_joint@meta.data <- seurat_joint@meta.data[, c(colnames(seurat_joint@meta.data)[1:22],
                                                     "Status_normal_malignant",
                                                     "Cell_type_projection",
                                                     "Cell_type_broad_Fig5")]

# clean up
rm(integrated_g34)
rm(ScSnDataWithNormals)

save(seurat_joint, file = glue("{out}/integrated_seurat.Rda"))

```

```{r load_v2, cache = FALSE}

load(glue("{out}/integrated_seurat.Rda"))

```

Subset the same signatures used in the normal brain to the genes detected in the tumors:

```{r signatures_keep}

interneuron_genes_keep <- base::intersect(interneuron_genes, rownames(seurat_joint@data))
astro_genes_keep <- base::intersect(astro_genes, rownames(seurat_joint@data))

```

Calculate mean expression:

```{r mean_exp_tumors}

data_tumors <- data.frame(
  interneuron_signature = cytobox::meanGeneExpression(seurat_joint, interneuron_genes_keep)$Mean_marker_expression,
  astrocyte_signature   = cytobox::meanGeneExpression(seurat_joint, astro_genes_keep)$Mean_marker_expression,
  cluster_broad         = seurat_joint@meta.data$Cell_type_broad_Fig5
)

```


Plot:

```{r tumor_scatter, fig.width = 6, fig.height = 5, echo_fig = TRUE}

data_tumors %>% 
  filter(cluster_broad %in% c("neurons", "astro")) %>% 
  rr_ggplot(aes(x = interneuron_signature, y = astrocyte_signature, colour = cluster_broad), plot_num = 1) +
  geom_point(shape = 16, size = 1, alpha = 0.2) +
  scale_color_manual(values = c("astro" = "#FF3F38", "neurons" = "#084081"), name = "Cell type") +
  geom_density_2d() +
  ggtitle("G34R/V HGGs")

```



<!-- END MATTER, insert reproducibility info -->

```{r footer, echo = FALSE, results = 'asis', warning = FALSE, cache = FALSE}

# Knit child document with header
res <- knitr::knit_child(here("include", "footer.Rmd"), envir = environment(), quiet = TRUE)
cat(res, sep = '\n')

```

<!-- END OF END MATTER -->
