---
title: "03 - Epigenome / ChIP-seq analysis"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  html_document:
    keep_md: true
    theme: flatly
    css: ../../include/style.css
    toc: yes
    toc_depth: 4
    number_sections: true
    df_print: paged
    includes:
      before_body: ../../include/header.html
      after_body:  ../../include/footer.html
---

<!-- FRONT MATTER, insert configuration info -->

```{r header, echo = FALSE, results = 'asis', warning = FALSE}

# Index of the document
# ...determines name of the subfolder of `outputs` and `figures`
doc_id <- "03"
subdir <- "bulk_transcriptome_epigenome"

suppressMessages(library(here))

# Knit child document with header
res <- knitr::knit_child(here("include", "header.Rmd"), envir = environment(), quiet = TRUE)
cat(res, sep = '\n')

```

<!-- END OF FRONT MATTER -->


# Overview

Here we take the G34 patient tumor epigenomic data (genic promoter scores for H3K27me3 and H3K27ac) and cross these values with the differentially-expressed genes by RNAseq, to identify genes with concordant differences
at the transcriptomic and epigenomic level.


# Libraries

```{r libs, warning = FALSE, message = FALSE, echo = TRUE, cache = FALSE}

# General
library(tidyr)
library(readr)
library(dplyr)
library(magrittr)
library(glue)

# For plotting
library(ggplot2)
library(RColorBrewer)
library(ggrepel)
ggplot2::theme_set(theme_min(base_size = 13))

# Custom
source(here::here(subdir, "analysis/functions.R"))

```

# Analysis

## Load data

Load tumor data:

```{r load_data}

dge <- read_tsv(here(subdir, "output/01/DGE_HGG-IDHvsHGG-G34R.V_padj<0.01_baseMean>100_absLFC>2.tsv")) %>% 
  separate(ID, into = c("ENSID", "symbol"), sep = ":")

```

Load and tidy epigenomics data:

```{r load_epi}

epi <- read_tsv(here(subdir, "data/20191112.RefSeq.K27me3K27ac.txt"))

```


Clean up gene names converted to dates:

```{r tidy_gene_names}

epi %>% distinct(name) %>% pull(name) %>% sort() %>% head(25)

epi <- epi %>%
  mutate(name = case_when(
    name == "01-Mar" ~ "MARCH1",
    name == "02-Mar" ~ "MARCH2",
    name == "03-Mar" ~ "MARCH3",
    name == "04-Mar" ~ "MARCH4",
    name == "05-Mar" ~ "MARCH5",
    name == "06-Mar" ~ "MARCH6",
    name == "07-Mar" ~ "MARCH7",
    name == "08-Mar" ~ "MARCH8",
    name == "09-Mar" ~ "MARCH9",
    name == "10-Mar" ~ "MARCH10",
    name == "11-Mar" ~ "MARCH11",
    name == "01-Sep" ~ "SEPT1",
    name == "02-Sep" ~ "SEPT2",
    name == "03-Sep" ~ "SEPT3",
    name == "04-Sep" ~ "SEPT4",
    name == "05-Sep" ~ "SEPT5",
    name == "06-Sep" ~ "SEPT6",
    name == "07-Sep" ~ "SEPT7",
    name == "08-Sep" ~ "SEPT8",
    name == "09-Sep" ~ "SEPT9",
    name == "10-Sep" ~ "SEPT10",
    name == "11-Sep" ~ "SEPT11",
    name == "12-Sep" ~ "SEPT12",
    name == "14-Sep" ~ "SEPT14",
    name == "01-Dec" ~ "DEC1",
    TRUE ~ name))

```

```{r epi_mean}

epi_mean <- epi %>%
  select(1:7, G34RV.K27me3.median, nonG34RV.K27me3.median, G34RV.K27ac.median, nonG34RV.K27ac.median, G34R.zK27m3, G34R.zK27ac) %>% 
  gather(Stat, Score, 8:ncol(.)) %>% 
  group_by(name, Stat) %>% 
  # Take the mean over different transcripts (although note that for each transcript,
  # the score is the same anyway)
  summarize(Score = mean(Score)) %>% 
  spread(Stat, Score) %>% 
  ungroup()

epi_indiv <- epi %>% 
  select(name, 8:40)

epi_meta <- data.frame(Sample = colnames(epi)[8:42],
                       Mark = c(rep("K27me3", 18), rep("K27Ac", 17)),
                       Group = c(rep("G34R/V", 5), rep("IDH/SETD2", 7), rep("WT", 6),
                                 rep("G34R/V", 6), rep("IDH/SETD2", 6), rep("WT", 5)))

rr_saveRDS(desc = "Median and z-scored values of promoter H3K27me3/ac for G34 and non-G34 entities",
           object = epi_mean,
           file = glue("{out}/genic_promoters_histone_summary.Rds"))

```

## Crossing epigenomics with other data

### DGE

```{r dge_and_epi, echo_fig = TRUE, fig.width = 7, fig.height = 5}

# Join the differentially expressed genes table and the epigenomics data
# table based on gene symbol
dge_and_epi <- dge %>%
  left_join(epi_mean, by = c("symbol" = "name")) %>% 
  arrange(abs(log2FoldChange)) %T>% 
  rr_write_tsv(path = glue("{out}/DGE_and_histone_marks.tsv"),
               desc = "Integrated RNA-seq differential expression and histone mark ChIPseq values for patient tumor samples")

dge_and_epi %>% 
  ggplot(aes(x = G34R.zK27ac, y = G34R.zK27m3)) +
  geom_hline(yintercept = 0.5, size = 0.5, colour = "gray90") +
  geom_vline(xintercept = 0.5, size = 0.5, colour = "gray90") +
  geom_point(aes(colour = log2FoldChange, size = -log10(padj)), alpha = 0.8) +
  scale_colour_gradientn(colours = colorRampPalette(c("navy", "white", "red"))(n = 200)) +
  cowplot::theme_cowplot()

goi <- dge_and_epi %>% filter(G34R.zK27ac > 0.9 | G34R.zK27m3 > 0.9) %>%
  filter(abs(log2FoldChange) > 3) %>%
  pull(symbol)

dge_and_epi %>% 
  rr_ggplot(aes(x = G34R.zK27ac, y = G34R.zK27m3), plot_num = 2) +
  geom_hline(yintercept = 0.5, size = 0.5, colour = "gray90") +
  geom_vline(xintercept = 0.5, size = 0.5, colour = "gray90") +
  geom_point(aes(colour = log2FoldChange, size = -log10(padj)), alpha = 0.8) +
  scale_colour_gradientn(colours = colorRampPalette(c("navy", "white", "red"))(n = 200)) +
  ggrepel::geom_label_repel(data = dge_and_epi %>% filter(symbol %in% goi), aes(label = symbol))

```


<!-- END MATTER, insert reproducibility info -->

```{r footer, echo = FALSE, results = 'asis', warning = FALSE, cache = FALSE}

# Knit child document with header
res <- knitr::knit_child(here::here("include", "footer.Rmd"), envir = environment(), quiet = TRUE)
cat(res, sep = '\n')

```

<!-- END OF END MATTER -->
