---
title: "04 - isogenic cell lines"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  html_document:
    keep_md: true
    theme: flatly
    css: ../../include/style.css
    toc: yes
    toc_depth: 4
    number_sections: true
    df_print: paged
    includes:
      before_body: ../../include/header.html
      after_body:  ../../include/footer.html
---

<!-- FRONT MATTER, insert configuration info -->

```{r header, echo = FALSE, results = 'asis', warning = FALSE}

# Index of the document
# ...determines name of the subfolder of `outputs` and `figures`
doc_id <- "04"
subdir <- "bulk_transcriptome_epigenome"

suppressMessages(library(here))

# Knit child document with header
res <- knitr::knit_child(here("include", "header.Rmd"), envir = environment(), quiet = TRUE)
cat(res, sep = '\n')

```

<!-- END OF FRONT MATTER -->


# Overview

In this document, we analyze bulk RNA-seq data from the G34-mutant patient-derived cell line GMB002 and isogenic matches, in which CRISPR editing was used to remove the mutation. Cell lines were also subjected to a differentiation protocol.


# Libraries

```{r libs, warning = FALSE, message = FALSE, echo = TRUE, cache = FALSE}

# General
library(tidyr)
library(readr)
library(readxl)
library(dplyr)
library(magrittr)
library(glue)
library(purrr)

# For plotting
library(ggplot2)
library(scales)
library(ggrepel)
library(cowplot)
ggplot2::theme_set(theme_min(base_size = 13))

# Custom
source(here::here(subdir, "analysis/functions.R"))
source(here::here(subdir, "analysis/ssgsea.R"))

```

# Analysis

## Prep bulk RNAseq analysis

### Sample metadata

Load in the metadata table for all bulk RNAseq samples to be included
in the analysis:

```{r bulk_samples}

(cell_line_meta <- readxl::read_xlsx(here(subdir, "data/20200219-cell_line_metadata.xlsx")) %>% 
     mutate(Group_broad = paste0(Genotype_broad, "_", Media)) %>% 
     mutate(Group_broad = gsub("/", "", Group_broad)))

# Number of samples per group
table(cell_line_meta$Group)

```

### Prepare bulk RNA-seq pipeline input

Palettes for groups:

```{r palette}

palette_groups <- c("G34R_ser" = "cyan4",
                    "G34R_nsm" = "cyan1",
                    "G34V_ser" = "cyan4",
                    "G34V_nsm" = "cyan1",
                    "KORepair_ser" = "navy",
                    "KORepair_nsm" = "blue")

save(palette_groups, file = glue("{out}/cl_palettes.Rda"))

```

We will need one version with all samples together, since normalization
is dataset-dependent, and we may want to compare expression levels across groups.

```{r info_samples_comp, dependson = 'palette'}

dir.create(file.path(out, "All_cell_line_data"), showWarnings = FALSE)

cell_line_meta %>%
    select(ID, Nickname, Group = Group_broad) %>%
    write_tsv(file.path(out, "All_cell_line_data", "info.samples.tsv"))

data.frame(Group = unique(cell_line_meta$Group_broad),
           Label = unique(cell_line_meta$Group_broad),
           Order = seq_along(unique(cell_line_meta$Group_broad)),
           Color = palette_groups[unique(cell_line_meta$Group_broad)],
           stringsAsFactors = FALSE) %>%
    write_tsv(file.path(out, "All_cell_line_data", "info.groups.tsv"))

```


## Run in-house bulk RNAseq pipeline

In order to run the in-house DESeq2/expression analysis pipeline, the inputs
created above are synced to the pipeline folder, and then
the following outputs are loaded for further analysis below and in the subsequent analyses:

- Alignment stats
- Differential gene expession analysis output
- Counts

```{r rp}

pipeline_path <- "../../../level3/2020-01_G34_submission1_add_samples/Cell_lines3/"

```

## Assemble RNAseq alignment stats

```{r align_stats}

align_stats <- read_tsv(file.path(pipeline_path, "All_cell_line_data", "alignment.statistics.tsv")) %>%
    separate(mitochondrialPercentage, sep = "%", into = "mitochondrialPercentage") %>%
    mutate(mitochondrialPercentage = as.numeric(mitochondrialPercentage))

# DT::datatable(align_stats, filter = "top", class = 'white-space: nowrap')

left_join(cell_line_meta, align_stats, by = c("Nickname" = "name")) %>%
    write_tsv(glue("{out}/cell_line_metadata_with_align_stats.tsv"))

```

## G34R vs KO / repair in serum

### Load DGE

```{r dge_serum}

dge_serum <- read_tsv(file.path(pipeline_path, "GBM002_ser/diff/Ensembl.ensGene.exon/G34R_servsKORepair_ser.tsv")) %>% 
    separate(ID, into = c("ENSID", "gene_symbol"), sep = ":") %>% 
    mutate(log10p = -log10(padj))

dge_serum %>% 
    filter(baseMean > 100 & padj < 0.05) %>% 
    rr_ggplot(aes(x = log2FoldChange, y = log10p, label = gene_symbol), alpha = 0.8, plot_num = 1) +
    geom_point()

dge_stem <- read_tsv(file.path(pipeline_path, "GBM002_nsm/diff/Ensembl.ensGene.exon/G34R_nsmvsKORepair_nsm.tsv")) %>% 
    separate(ID, into = c("ENSID", "gene_symbol"), sep = ":") %>% 
    mutate(log10p = -log10(padj))

```


### Targeted DGE, for serum condition - GBM002

Note that here, we load the counts which were normalized together for serum
samples only, since the normalization by DESeq2 is sample-dependent.

```{r targeted_dge_serum only, fig.width = 14, fig.height = 5}

goi <- c("DLX1", "DLX2", "DLX5", "DLX6", "PDGFRA", "GSX2")

# see function specification in functions.R
gbm002_counts <- extract_pipeline_counts(file.path(pipeline_path, "GBM002_ser/counts/Ensembl.ensGene.exon.norm.tsv.gz"),
                                         goi = goi) %>%
    left_join(cell_line_meta, by = c("sample" = "Nickname")) %>% 
    left_join(dge_serum, by = c("gene_symbol" = "gene_symbol", "gene_ensg" = "ENSID"))

dge_serum %>% select(-ENSID) %>% filter(gene_symbol %in% goi)

```

For the DLXs only

```{r dlx_serum_boxplot, fig.width = 14, fig.height = 5, echo_fig = TRUE}

dlx_limits <- list("DLX1"   = c(0, 21000),
                   "DLX2"   = c(0, 21000),
                   "DLX5"   = c(0, 1600),
                   "DLX6"   = c(0, 1600),
                   "PDGFRA" = c(0, 22000),
                   "GSX2"   = c(0, 200))

cl_dotplot <- function(gene, lims, counts, levels = c("G34R_nsm", "KORepair_nsm",
                                                      "G34R_ser", "KORepair_ser")) {
    
    counts %>%
        filter(gene_symbol == gene) %>% 
        mutate(Group_broad = factor(Group_broad,
                                    levels = levels)) %>% 
        ggplot(aes(x = Group_broad, y = gene_expression)) +
        geom_jitter(aes(colour = Group_broad), size = 5, alpha = 0.87, width = 0.1) +
        scale_colour_manual(values = c("cyan4", "midnightblue")) +
        stat_summary(fun.y = median, fun.ymin = median, fun.ymax = median,
                     geom = "crossbar", width = 0.2) +
        theme_min(border_colour = "black") +
        ylim(lims) +
        ggtitle(gene) +
        rotateX() +
        noLegend()
    
}

gbm002_counts %>% 
    select(sample, gene_symbol, gene_expression, Group_broad) %>% 
    write_tsv(glue("{figout}/dlx_serum_boxplot-1.source_data.tsv"))

imap(dlx_limits, ~ cl_dotplot(.y, .x, counts = gbm002_counts)) %>% 
    {plot_grid(plotlist = ., ncol = 6)}

```

#### Investigation of DGE in the serum condition

```{r ma_plot}

# Load up DGE results from DESeq2
res_ser <- read_tsv(file.path(pipeline_path, "GBM002_ser/diff/Ensembl.ensGene.exon/G34R_servsKORepair_ser.tsv")) %>% 
    separate(ID, into = c("ENSG", "symbol"), sep = ":") 

# Construct the input dataframe for an MA plot
res_ser2 <- res_ser %>%
    select(symbol, baseMean, log2FoldChange, padj) %>%
    mutate(signif   = ifelse( is.na(padj), FALSE, padj < 0.05 )) %>% 
    mutate(dlx      = symbol %in% c("DLX1", "DLX2", "DLX5", "DLX6")) %>% 
    mutate(fc = case_when(
        log2FoldChange > 2 ~ "LFC > 2",
        log2FoldChange < -2 ~ "LFC < -2",
        TRUE ~ "-2 < LFC < 2")) %>% 
    mutate(log2FoldChange = ifelse(abs(log2FoldChange) > 2,
                                   sign(log2FoldChange) * 2,
                                   log2FoldChange))

# Make an MA plot, labelling the DLX genes
res_ser2 %>%
    ggplot(aes(x = log10(baseMean), y = log2FoldChange)) +
    geom_hline(yintercept = 0, colour = "gray90") +
    geom_point(aes(colour = signif, size = dlx, shape = fc), alpha = 0.5) +
    scale_colour_manual(values = c("gray50", "red")) +
    scale_size_manual(values = c("TRUE" = 3, "FALSE" = 0.5)) +
    scale_shape_manual(values = c("LFC > 2" = 24, "LFC < -2" = 25, "-2 < LFC < 2" = 19)) +
    geom_text_repel(data = res_ser2 %>% filter(symbol %in% c("DLX1", "DLX2", "DLX5", "DLX6")),
               aes(x = log10(baseMean), y = log2FoldChange, label = symbol)) +
    ylim(c(-2, 2))

```

Examination of p-values:

```{r p-values}

# Here's a histongram of p-values across genes
res_ser %>% 
    gather(stat, value, pvalue, padj) %>%
    ggplot(aes(x = value)) +
    geom_histogram() +
    facet_wrap(~ stat, ncol = 1)

# Filter to genes with baseMean > 100 and redraw the plot
res_ser %>% 
    filter(baseMean > 100) %>% 
    gather(stat, value, pvalue, padj) %>%
    ggplot(aes(x = value)) +
    geom_histogram() +
    facet_wrap(~ stat, ncol = 1)

```

### Targeted DGE, for stem condition - GBM002

```{r targeted_dge_nsm}

# see function specification in functions.R
gbm002_counts_nsm <- extract_pipeline_counts(file.path(pipeline_path, "GBM002_nsm/counts/Ensembl.ensGene.exon.norm.tsv.gz"),
                                             goi = goi) %>%
    left_join(cell_line_meta, by = c("sample" = "Nickname")) %>% 
    left_join(dge_stem, by = c("gene_symbol" = "gene_symbol", "gene_ensg" = "ENSID"))

dge_stem %>% select(-ENSID) %>% filter(gene_symbol %in% goi)

```

```{r dlx_stem_boxplot, fig.width = 14, fig.height = 5, echo_fig = TRUE}

dlx_limits <- list("DLX1"   = c(0, 4000),
                   "DLX2"   = c(0, 5300),
                   "DLX5"   = c(0, 400),
                   "DLX6"   = c(0, 500),
                   "PDGFRA" = c(0, 30000),
                   "GSX2"   = c(0, 30))

gbm002_counts_nsm %>% 
    select(sample, gene_symbol, gene_expression, Group_broad) %>% 
    write_tsv(glue("{figout}/dlx_stem_boxplot-1.source_data.tsv"))

imap(dlx_limits, ~ cl_dotplot(.y, .x, counts = gbm002_counts_nsm)) %>% 
    {plot_grid(plotlist = ., ncol = 6)}

```


## ssGSEA

### Prep counts

```{r raw_counts}

raw_counts <- file.path(pipeline_path, "All_GBM002_new/counts/Ensembl.ensGene.exon.raw.tsv.gz") %>%
    read.table(header = T, row.names = 1, check.names = F, sep = "\t")

head(rownames(raw_counts) %<>% strsplit(":") %>% sapply(getElement, 1))
raw_counts <- as.matrix(raw_counts)

raw_counts[1:5, 1:5]
rr_saveRDS(object = raw_counts,
           desc = "Raw RNA-seq counts for GBM002 cell lines",
           file = glue("{out}/cl_raw_counts.Rds"))

dim(raw_counts)

```


### Load signatures

Loading the cell type gene signatures prepared previously for GSEA, and selecting
some representative signatures for relevant cell types:

```{r load_sigs}

signatures <- readRDS(here(subdir, "output/02/signatures_ens.Rds"))

# Select pathways of interest
poi <- list("newborn_inhib_neuron" = signatures$`HF nIN1`,
            "mature_inhib_neuron" = signatures$`HP IN-PV`,
            "opc" = signatures$`F-p6 OPC`,
            "astrocyte" = signatures$`F-p0 ASTR2`,
            "rgc" = signatures$`HF RG-early`,
            "fetal_excit_neurons" = signatures$`HF EN-PFC1`)

```

### Run ssGSEA

```{r run_ssgsea, results = "hide", eval = TRUE}

cl_ssgsea <- ssgsea_le(expr_mat = raw_counts,
                       gene_sets = poi,
                       # Following exactly the same parameters as we used
                       # in the bulk RNAseq analysis for Jessa et al, Nature Genetics, 2019
                       alpha = 0.75,
                       normalize = FALSE)

cl_ssgsea$enrichment_scores %>%
    as.data.frame() %>%
    tibble::rownames_to_column(var = "Signature") %>%
    rr_write_tsv(path = glue("{out}/cl_ssgsea_scores.tsv"),
                 desc = "ssGSEA scores for cell type signatures in GBM002 cell lines")

leading_edge <- cl_ssgsea$leading_edge
rr_saveRDS(object = leading_edge,
           desc = "Leading edge froms ssGSEA analysis of cell type signatures in GBM002 cell lines",
           file = glue("{out}/cl_ssgsea_le.Rds"))

```

### Visualize results

```{r viz_ssgsea, width = 2.5, height = 3, echo_fig = TRUE}

cl_ssgsea_tidy <- cl_ssgsea$enrichment_scores %>%
    as.data.frame() %>%
    tibble::rownames_to_column(var = "Signature") %>%
    gather(Sample, Score, 2:ncol(.)) %>%
    left_join(cell_line_meta, by = c("Sample" = "Nickname"))

cl_ssgsea_tidy %>%
    filter(Signature %in% c("newborn_inhib_neuron", "rgc")) %>%
    mutate(Signature = factor(Signature, levels = c("rgc", "newborn_inhib_neuron"))) %>%
    filter(Media == "ser") %>%
    mutate(Group_broad = factor(Group_broad,
                                levels = c("G34R_nsm", "KORepair_nsm",
                                           "G34R_ser", "KORepair_ser"))) %>%
    rr_ggplot(aes(x = Group_broad, y = Score), plot_num = 1) +
    geom_jitter(aes(colour = Group_broad), size = 5, alpha = 0.87, width = 0.1) +
    scale_colour_manual(values = c("cyan4", "midnightblue")) +
    stat_summary(fun.y = median, fun.ymin = median, fun.ymax = median,
                 geom = "crossbar", width = 0.2) +
    theme_min(border_colour = "black") +
    facet_wrap(~ Signature) +
    rotateX() +
    noLegend() +
    ylim(c(22400, 24000))

```

## Distinguishing the effect of G34 vs. cell of origin

In a comparison of H3K27me3 at promoters in G34-mutant HGG patient tumor samples vs non-G34 mutant
HGGs, G34 tumors have several thousand promoters enriched for K27me3 compared
to the other tumor types.

To investigate whether these are an effect of the G34 mutation, or an effect
of cell of origin (both of which differ between patient tumors), we'll look
at the expression of these promoters in the CRISPR cell lines, allowing to decouple
the effect of G34 (which differs in the isogenic cell line pairs) vs. cell of origin (which
does not).

```{r read_tumor_k27me3}

# Read in the tumour promoter K27me3 scores
tumor_k27me3 <- read_xlsx(here(subdir, "data/TABLE-S3_G34-HGG-Epigenome.xlsx"))

# Subset by thresholds used in the paper
# z-score_H3K27me3 > 0.5, pvalue_H3K27me3 < 0.05
tumor_k27me3_filt <- tumor_k27me3 %>% 
    filter(`z-score_H3K27me3` > 0.5) %>% 
    distinct(name, .keep_all = TRUE)

# Read in the counts for the cell lines - see function specification 
# in functions.R
gbm002_counts <- extract_pipeline_counts(file.path(pipeline_path, "GBM002_ser/counts/Ensembl.ensGene.exon.norm.tsv.gz"),
                                         goi = tumor_k27me3_filt$name) %>%
    left_join(cell_line_meta, by = c("sample" = "Nickname"))

# Check fold changes from RNA-seq DGE in stem at these promoters
dge_stem %>% 
    filter(gene_symbol %in% tumor_k27me3_filt$name) %>% 
    ggplot(aes(x = log2FoldChange)) +
    geom_density()

# Check what proportion of those genes have associated significant DGE
dge_stem %>%
    filter(gene_symbol %in% tumor_k27me3_filt$name) %>% 
    mutate(significant_upon_editing = case_when(
        is.na(padj) ~ "No",
        padj < 0.05 & log2FoldChange > 1 ~ "Yes",
        TRUE ~ "No")) %>%
    count(significant_upon_editing) %>% 
    ggplot(aes(x = "", y = n)) +
    geom_bar(aes(fill = significant_upon_editing), stat = "identity") +
    coord_polar("y", start = 0) +
    theme_void() +
    geom_text(aes(x = c(1, 1), y = c(1000, 0), label = n), size=5)

```



<!-- END MATTER, insert reproducibility info -->

```{r footer, echo = FALSE, results = 'asis', warning = FALSE, cache = FALSE}

# Knit child document with header
res <- knitr::knit_child(here::here("include", "footer.Rmd"), envir = environment(), quiet = TRUE)
cat(res, sep = '\n')

```

<!-- END OF END MATTER -->
